/* Copyright 2020 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";
import "tensorflow/core/framework/tensor.proto";
import "fedlearner/common/common.proto";
import "google/protobuf/any.proto";

package fedlearner.common;

message TransmitRequest {
  uint64 iter_id = 1;
  string name = 2;
  tensorflow.TensorProto tensor = 3;
  google.protobuf.Any any_data = 4;
};

message TransmitResponse {
  Status status = 1;
};

message LoadDataBlockRequest {
  uint64 count = 1;
  string block_id = 2;
};

message LoadDataBlockResponse {
  Status status = 1;
};

service TrainerWorkerService {
  rpc StreamTransmit (stream TransmitRequest) returns (stream TransmitResponse) {}
  rpc LoadDataBlock (LoadDataBlockRequest) returns (LoadDataBlockResponse) {}
}
